certstore/snapshot.go

```diff
@@ -0,0 +1,90 @@
+package certstore
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"errors"
+	"io"
+
+	"github.com/filecoin-project/go-f3/gpbft"
+	"github.com/ipfs/go-datastore"
+	xerrors "golang.org/x/xerrors"
+)
+
+var ErrlatestCertificateNil = errors.New("latest certificate is not available")
+
+// Exports an F3 snapshot that includes the finality certificate chain until the current `latestCertificate`.
+func (cs *Store) ExportLatestSnapshot(ctx context.Context, writer io.Writer) error {
```

Comment:
The term `Snapshot` doesn't carry enough of a weight within the context of go-f3. What we really mean by it at this stage is simply: export certificates.

You _could_ make it mean something: define a type `Snapshot` that is either produced by the `Store`, or takes the `Store`, and implements `io.WriterTo`, `io.ReaderFrom` etc.

Or avoid the term entirely and instead make the store simply an `Iterator` of certs and separate IO ops elsewhere. 

I would probably go with the first approach but there is not enough implementation in this PR for me to fully wrap my head around how you are thinking of approaching the problem. 

So please feel free to ignore the recommendations if the happen to not make sense as the work progresses forward.

Reply:
Added FRC link to the comment. As I understand, a `Snapshot` struct is an in-memory representation of the snapshot which is not ideal for implementing a streaming export with a low RAM usage. Implementing `io.WriterTo` would not allow cancellation with `context.Context`.